pkg/migrate/migrate.go

```diff
@@ -330,7 +347,30 @@ func copyOnePVC(ctx context.Context, w *log.Logger, clientset k8sclient.Interfac
 	return nil
 }
 
-func createMigrationPod(ctx context.Context, clientset k8sclient.Interface, ns string, sourcePvcName string, destPvcName string, rsyncImage string) (*corev1.Pod, error) {
+func createMigrationPod(ctx context.Context, clientset k8sclient.Interface, ns string, sourcePvcName string, destPvcName string, rsyncImage string, nodeName string) (*corev1.Pod, error) {
+
+	// apply nodeAffinity when migrating to a local volume provisioner
```

Comment:
why not just _always_ apply affinity?

Reply:
Assuming we know what the required node would be

Reply:
why not make it work for things like local volume provisioner/hostpath/whatever at the same time?

Reply:
There are certain cases where I can't just apply a nodeAffinity because the program can't determine the nodeName for a pvc consumer. E.g. prometheus pods - they are managed by the operator and are outside the program's control.

Reply:
lets not change anything now for other provisioners as that will increase the surface area of the change and force us to run manual tests with longhorn. we can follow up with a story and this change after we ship this feature.

Reply:
Internal tracking [issue](https://app.shortcut.com/replicated/story/61150/pv-migrate-followup-how-to-handle-scheduling-of-pods-after-a-pvc-migration).